### PR TITLE
Test cluster status Available and Failing

### PR DESCRIPTION
**Pull request by kevinrizza**
Problem:
Currently, the e2e extended test "[Feature:Platform][Smoke]
Managed cluster should start all core operators" only fails if the timer
finishes and any of the core operators have status `Available=="False"`.
However, it is possible for `Available` to be true and `Degraded` to also
be set to true.

In general this case of `Available` set to true and `Degraded` set to
true should prevent this test from passing because cluster resources
should not be in a failing state by default. However in the case of the
`marketplace` core operator, an external dependency (currently to an api
hosted on quay.io) is needed in order to handle default operands.
In general, this test shouldn't fail due to an external dependency.

Solution:
Update the test to fail when the cluster status has `Available` set to
true and `Degraded` set to true. Additionally, create a whitelist so that
marketplace (and any future operators that have external dependencies)
can ignore this particular failure. Additionally, e2e tests for the
marketplace operator exist that will fail if that external api (from
quay.io) is unavailable.

### CODE REVIEW

**Comment by mfojtik** [comment]
Failing is now Degraded

File: test/extended/operators/operators.go [hunk 1]
```diff
@@ -203,6 +211,16 @@ var _ = g.Describe("[Feature:Platform] Managed cluster should", func() {
 	})
 })
 
+func inAvailableAndFailingWhitelist(coName string) bool {
+	whitelist := map[string]struct{}{
+		"marketplace": {},
```
Comment (wking) [inline test/extended/operators/operators.go]: If you're going to inline the map (vs. keeping it in an external variable), you might as well use a `switch`:

```go
switch coName {
case "marketplace": return true
default: return false
```

**Comment by wking** [comment]
> Failing is now Degraded

Modulo this change, e3013ba looks good to me.

File: test/extended/operators/operators.go (hunk 1 above)
Comment (kevinrizza) [reply test/extended/operators/operators.go]: > Failing is now Degraded

@mfojtik So are you saying that this test as written is wrong? It seems to me like the current state of ClusterOperator still has the Failing field. For example, all of the clusteroperator code in the openshift-marketplace operator still refers to it as Failing.

Or are you saying that this change is in progress? How is something like that going to get communicated to operator owners that already write that field? If it is merged, can you point me to a commit?

**Comment by kevinrizza** [comment]
> Failing is now Degraded

@mfojtik So are you saying that this test as written is wrong? It seems to me like the current state of ClusterOperator still has the Failing field. For example, all of the clusteroperator code in the openshift-marketplace operator still refers to it as Failing.

Or are you saying that this change is in progress? How is something like that going to get communicated to operator owners that already write that field? If it is merged, can you point me to a commit?

File: test/extended/operators/operators.go (hunk 1 above)
Comment (kevinrizza) [reply test/extended/operators/operators.go]: Fair point. Updated in latest push.

**Comment by wking** [comment]
> ... ClusterOperator still has the Failing field...

See openshift/api@a9fb3b1629b23b3198c3d94017fc63430afcf6cc

**Comment by kevinrizza** [comment]
@mfojtik @wking @awgreene @shawn-hurley 

Bringing this one back from the dead. Updated to reference the correct fields in the ClusterOperator status (replaced failing with degraded)

Please take another look, as this pr now blocks https://github.com/operator-framework/operator-marketplace/pull/235

**Comment by awgreene** [comment]
@mfojtik would you mind reviewing @kevinrizza's latest changes?

**Comment by kevinrizza** [comment]
/hold

going to wait on this one until we come up with a better strategy.

**Pull request closed without merging**